### PR TITLE
more suppression of logging

### PR DESCRIPTION
**Pull request by sethalves**
- more suppression of logging

https://highfidelity.fogbugz.com/f/cases/14488/Implement-Quite-Logging

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9553/app_13114_PR_e24dd2166343eb8f823c47b82833e50f86712952.apk

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9556/app_13114_PR_e24dd2166343eb8f823c47b82833e50f86712952.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9556/HighFidelity-Beta-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9556/HighFidelity-Beta-Interface-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.exe

**Comment by sethalves** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9557/HighFidelity-Beta-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9557/HighFidelity-Beta-Interface-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9557/HighFidelity-Beta-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9557/HighFidelity-Beta-Interface-PR13114-e24dd2166343eb8f823c47b82833e50f86712952.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2247,6 +2266,24 @@ void Application::updateVerboseLogging() {
...
+    const_cast<QLoggingCategory*>(&gpulogging())->setEnabled(QtInfoMsg, enable);
+
+    const_cast<QLoggingCategory*>(&gpugllogging())->setEnabled(QtDebugMsg, enable);
+    const_cast<QLoggingCategory*>(&gpugllogging())->setEnabled(QtInfoMsg, enable);
+
+    const_cast<QLoggingCategory*>(&qmlLogging())->setEnabled(QtDebugMsg, enable);
+    const_cast<QLoggingCategory*>(&qmlLogging())->setEnabled(QtInfoMsg, enable);
+
+    const_cast<QLoggingCategory*>(&entities())->setEnabled(QtDebugMsg, enable);
+    const_cast<QLoggingCategory*>(&entities())->setEnabled(QtInfoMsg, enable);
```
Comment (Atlante45) [inline interface/src/Application.cpp]: I'm really not a fan of casting away const on a static const object here.
There's also a proper way to do this that does not require to do it for every single category.
Using this:
```cpp
QString rules = "*.debug=%1\n"
                "*.info=%1";
rules = rules.arg(enable ? "true" : "false");
QLoggingCategory::setFilterRules(rules);
```

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -2207,10 +2214,16 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
     _pendingRenderEvent = false;
 
     qCDebug(interfaceapp) << "Metaverse session ID is" << uuidStringWithoutCurlyBraces(accountManager->getSessionID());
+
+    updateVerboseLogging();
```
Comment (Atlante45) [inline interface/src/Application.cpp]: Why call it twice in the ctor?

**Comment by sethalves** [comment]
to test:
- run interface
- open log
- note that after the initial burst, logging no longer includes [debug] or [info] categories

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9642/app_13114_PR_16e832ac0894a2c2413dbdc3328da148757c00d0.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9642/HighFidelity-Beta-PR13114-16e832ac0894a2c2413dbdc3328da148757c00d0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9642/HighFidelity-Beta-Interface-PR13114-16e832ac0894a2c2413dbdc3328da148757c00d0.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9642/HighFidelity-Beta-PR13114-16e832ac0894a2c2413dbdc3328da148757c00d0.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9642/HighFidelity-Beta-Interface-PR13114-16e832ac0894a2c2413dbdc3328da148757c00d0.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Logging is significantly less intrusive.

**Comment by conklin94122** [comment]
@Atlante45 @sethalves Were Clement's requests addressed already or are they still pending?

File: interface/src/Application.cpp (hunk 1 above)
Comment (Atlante45) [reply interface/src/Application.cpp]: If we do wat to do this in the future, we can always customize the filter to something like that:
```
*.debug=false
*.info=false
entities.debug=true
entities.info=true
````

**Pull request merged**